tests/optimizers/bohb_test.py

```diff
@@ -82,6 +87,36 @@ def test_bohb_parallel():
         assert es.optimizer_info["configuration_key"] == (1, 0, i)
 
 
+def test_bohb_report_as_batch():
+    paramspace = ps.ParameterSpace()
+    paramspace.add(ps.ContinuousParameter("p1", [0, 1]))
+    opt = BOHB(
+        paramspace,
+        Objective("loss", False),
+        min_fidelity=0.2,
+        max_fidelity=1,
+        num_iterations=1,
+    )
+
+    evaluations = []
+    for i in range(3):
+        es = opt.get_evaluation_specification()
+        evaluation = es.create_evaluation(objectives={"loss": i})
+        evaluations.append(evaluation)
+
+    # Add an evaluation _not_ created by the optimizer to see if it get's skipped
+    # and triggers a warning on reporting:
+    invalid_evaluation = Evaluation(objectives={"loss": 42}, configuration={})
+    evaluations.append(invalid_evaluation)
+
+    assert len(opt.pending_configurations) == 3
+    with pytest.raises(EvaluationsError) as excinfo:
+        opt.report(evaluations)
+    assert len(opt.pending_configurations) == 0
+    assert excinfo.value.message.startswith("An error with one or more evaluation")
+    assert excinfo.value.evaluations_with_errors[0][0] == invalid_evaluation
```

Comment:
How is this different from the matching reference test that BOHB is also subjected to?